tests/cases/fourslash/codeFixConvertTypedefToType4.ts

```diff
@@ -0,0 +1,30 @@
+/// <reference path='fourslash.ts' />
+
+////
+//// /**
+////  * @typedef {object} Person
+////  * @property {object} data
+////  * @property {string} data.name
+////  * @property {number} data.age
+////  * @property {object} data.contact
+////  * @property {string} data.contact.address
+////  * @property {string} data.contact.phone
```

Comment:
Can you test something like `[data.contact.address]`, which should produce an optional property?

Reply:
done